hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -209,6 +245,15 @@ public void persistToken(Token token) {
 		final var mutableToken = currentTokens.getForModify(key);
 		mapModelChangesToMutable(token, mutableToken);
 
+		if (token.hasMintedUniqueTokens()) {
+			for (UniqueToken uniqueToken : token.mintedUniqueTokens()) {
+				final var merkleUniqueId = new MerkleUniqueTokenId(new EntityId(uniqueToken.getTokenId()), uniqueToken.getSerialNumber());
+				final var merkleUniqueToken = new MerkleUniqueToken(new EntityId(uniqueToken.getOwner()), uniqueToken.getMetadata(), uniqueToken.getCreationTime());
+				uniqueTokens.get().put(merkleUniqueId, merkleUniqueToken);
+			}
+			accountStore.persistAccount(token.getTreasury());
```

Comment:
I'm leaning toward a pattern in which a `TransitionLogic` must explicitly call a `*Store.persist*` method for **every** entity it expects to be changed in the transaction.

This does lead to some awkward idioms, but I like the single-method-visibility it gives to exactly what state changes will be caused by the transaction.